ethcore/snapshot/src/lib.rs

```diff
@@ -57,22 +55,29 @@ use state_db::StateDB;
 use trie_db::{Trie, TrieMut};
 
 pub use self::consensus::*;
-pub use self::service::Service;
-pub use self::traits::{SnapshotService, SnapshotComponents, Rebuilder};
+pub use self::service::{Service, Guard, Restoration, RestorationParams};
+pub use self::traits::{SnapshotService, SnapshotClient, SnapshotComponents, Rebuilder};
+pub use self::io::SnapshotWriter;
 pub use self::watcher::Watcher;
-pub use common_types::basic_account::BasicAccount;
+use common_types::basic_account::BasicAccount;
 
 pub mod io;
 pub mod service;
 
+#[cfg(any(test, feature = "test-helpers"))]
```

Comment:
well, I think `feature = "test-helpers"` should be enough because `snapshot` doesn't contain any tests after this PR but I don't mind if we keep it to be on safe-side.

Reply:
Also, I think we could have something like:

```rustc
#[cfg(not(any(test, feature = "test-helpers")))]
pub mod test_helpers {
   pub use watcher::Watcher;
    .... export everything we need
}
```
However, I don't know how painful it would be to change the import paths for the tests but it would reduce the `cfg stff` a bit

Reply:
It makes the tests a wee bit more opaque, but overall I think your suggestion is better than what I had. 👍